mapflow/mapflow.py

```diff
@@ -1226,6 +1235,26 @@ def calculate_aoi_area(self, aoi: QgsGeometry, crs: QgsCoordinateReferenceSystem
         self.calculator.setSourceCrs(helpers.WGS84, self.project.transformContext())
         self.aoi_size = self.calculator.measureArea(aoi) / 10 ** 6  # sq. m to sq.km
         self.dlg.labelAoiArea.setText(self.tr('Area: {:.2f} sq.km').format(self.aoi_size))
+        # get workflow def if for selected model
+        workflow_def_id = self.workflow_def_ids.get(
+            self.dlg.modelCombo.currentText()
+        )
+        self.calculate_processing_cost(aoi=aoi, workflow_def_id=workflow_def_id)
+
+    def calculate_processing_cost(self, aoi: QgsGeometry, workflow_def_id: uuid.UUID) -> None:
+        """:return: calculated aoi area and cost in credits"""
+        geometry = json.loads(aoi.asJson())
+        body = {"wdId": workflow_def_id,
+                "geometry": geometry}
+        self.http.post(
+            url=f"{self.server}/processing/cost",
+            callback=self.calculate_processing_cost_callback,
+            body=json.dumps(body).encode()
+        )
+
+    def calculate_processing_cost_callback(self, response: QNetworkReply):
+        response_data = response.readAll().data().decode()
+        self.dlg.labelAoiArea.setText(f'{self.dlg.labelAoiArea.text()}\n{response_data} credits')
```

Comment:
Wouldn't it be better to add separate label?

This function changes the current label value, but the initial value is set elswhere. What if there will be nothing, or the cost will already be there? Currently it is not so, because calculate_processing_cost is called in one place only, but later it can change, and it will be hard to maintain

Reply:
Also, do not forget to use self.tr() to add translations later!

Reply:
You're right, I will add a separate label.